File: VB/MyGridControl/MyGridViewInfoRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Registrator



Namespace DevExpress.MyControl
	Public Class MyGridViewInfoRegistrator
		Inherits GridInfoRegistrator
		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property



		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New MyGridView(TryCast(grid, GridControl))
		End Function
	End Class
End Namespace

File: CS/MyGridControl/MyGridViewInfoRegistrator.cs
using System;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Registrator;



namespace DevExpress.MyControl
{
    public class MyGridViewInfoRegistrator : GridInfoRegistrator
    {
        public override string ViewName 
        { 
            get { return "MyGridView"; } 
        }



        public override BaseView CreateView(GridControl grid) 
        { 
            return new MyGridView(grid as GridControl); 
        }
    }
}
